cross-media-measurement/src/test/kotlin/org/wfanet/measurement/integration/common/InProcessKingdomIntegrationTestBase.kt

```diff
@@ -171,8 +178,10 @@ abstract class InProcessKingdomIntegrationTestBase {
     // These states are exposed in GlobalComputation as CREATED and CONFIRMING.
     logger.info("Awaiting a GlobalComputation in CONFIRMING state")
     val computation = pollFor {
-      globalComputations.findLast {
-        it.state == GlobalComputation.State.CONFIRMING
+      globalComputationsMutex.withLock {
+        globalComputations.findLast {
+          it.state == GlobalComputation.State.CONFIRMING
+        }
```

Comment:
Since we do this twice, how about simplifying the core test code with this helper function:

```
fun findLastComputationInState(state: GlobalComputation.State): GlobalComputation {
  return pollfor { globalComputationsMutex.withLock { globalComputations.findLast { it.state == state } } }
}
```